data/archiver/archiver.py

```diff
@@ -1,40 +1,45 @@
-import os
 import logging
 from data.archiver.aws_s3_client import AwsS3
 from data.archiver.ftp_uploader import FtpUploader
+from data.archiver.ingest_api import Ingest
 from data.archiver.stream import S3FTPStreamer
 from data.archiver.utils import compress, md5
 from data.archiver.dataclass import DataArchiverRequest, DataArchiverResult, FileResult
-from data.archiver.ingest_api import Ingest
 
 
 class Archiver:
 
-    def __init__(self):
-        self.ingest_api = Ingest()
-        self.aws_client = AwsS3()
+    def __init__(self, ingest_cli: Ingest, aws_cli: AwsS3):
+        self.ingest_cli = ingest_cli
+        self.aws_cli = aws_cli
         self.ftp = None
         
         self.logger = logging.getLogger(__name__)
         self.logger.setLevel(logging.INFO)
 
     def start(self, req: DataArchiverRequest):
         self.logger.info(req)
-        self.logger.info(f'Getting sequence files for submission {req.sub_uuid} from INGEST_API')
-        sequence_files = self.ingest_api.get_sequence_files(req.sub_uuid)
+        self.logger.info(f'Getting sequence files for submission {req.sub_uuid} from Ingest.')
+        sequence_files = self.ingest_cli.get_sequence_files(req.sub_uuid)
         self.logger.info(sequence_files)
 
         if not sequence_files:
-            res = DataArchiverResult(req.sub_uuid, success=False, error='No sequence files in submission')
+            res = DataArchiverResult(req.sub_uuid, success=False, error='No sequence files in submission.')
             self.logger.info(res)
             return res
 
         if not req.files:
             # archive all files
-            res_files = list(map(lambda f: FileResult(f["file_name"], f["cloud_url"]), sequence_files))
+            res_files = list(map(lambda f: FileResult(f["uuid"], f["file_name"], f["cloud_url"]), sequence_files))
         else:
-            staging_area = self.ingest_api.get_staging_area()
-            res_files = list(map(lambda f: FileResult(f, staging_area + f), req.files))
+            def file_result(uuid):
+                for f in sequence_files:
+                    if f["uuid"] == uuid:
+                        return FileResult(f["uuid"], f["file_name"], f["cloud_url"])
+                return FileResult(uuid, None, None, success=False, error="File not found in Ingest.")
```

Comment:
Passing `None` arguments is unclean & makes it difficult to convey & understand intent.
Also here a factory method would simplify stuff.

```python
class FileResult:
    def not_found_error(uuid):
        return FileResult(uuid, None, None, success=False, error="File not found in Ingest.")
 ```